scilpy/tracking/seed.py

```diff
@@ -1,27 +1,43 @@
 # -*- coding: utf-8 -*-
+import logging
+
 import numpy as np
 
 
 class SeedGenerator(object):
     """
-    Class to get seeding positions
+    Class to get seeding positions.
+
+    Generated seeds are in voxmm space, origin=corner. Ex: a seed sampled
+    exactly at voxel i,j,k = (0,1,2), with resolution 3x3x3mm will have
+    coordinates x,y,z = (0, 3, 6).
+
+    Using get_next_pos, seeds are placed randomly within the voxel. In the same
+    example as above, seed sampled in voxel i,j,k = (0,1,2) will be somewhere
+    in the range x = [0, 3], y = [3, 6], z = [6, 9].
     """
-    def __init__(self, img):
+    def __init__(self, data, voxres):
         """
         Parameters
         ----------
-        img: nibabel image
-            The seeding mask. Seeds will be randomly placed in voxels with
-            value >0.
+        data: np.array
+            The data, ex, loaded from nibabel img.get_fdata().
+        voxres: np.array(3,)
+            The pixel resolution, ex, using img.header.get_zooms()[:3].
         """
-        self.pixdim = img.header.get_zooms()[:3]
+        self.data = data
+        self.voxres = voxres
 
-        data = img.get_fdata(caching='unchanged', dtype=np.float64)
-        self.seeds = np.array(np.where(np.squeeze(data) > 0)).transpose()
+        # self.seed are all the voxels where a seed could be placed
+        # (voxel space, int numbers).
+        self.seeds = np.array(np.where(np.squeeze(data) > 0),
+                              dtype=float).transpose()
+        if len(self.seeds) == 0:
+            logging.warning("There are positive voxels in the seeding mask!")
 
     def get_next_pos(self, random_generator, indices, which_seed):
         """
-        Generate the next seed position.
+        Generate the next seed position (Space=voxmm, origin=corner).
```

Comment:
We need to be careful with origin. It is `corner` here, but default is `center` in `DataVolume`.

Reply:
I will just remove the default in DataVolume. Nice catch.